src/llscan.cc

```diff
@@ -1202,14 +1270,77 @@ bool LLScan::ScanHeapForObjects(lldb::SBTarget target,
 
   /* Populate the map of objects. */
   if (mapstoinstances_.empty()) {
-    FindJSObjectsVisitor v(target, GetMapsToInstances());
+    FindJSObjectsVisitor v(target, this);
 
     ScanMemoryRanges(v);
   }
 
   return true;
 }
 
+std::unique_ptr<std::string>
+FindJSObjectsVisitor::MapCacheEntry::GetTypeNameWithProperties(
```

Comment:
You don't have to wrap it in a unique_ptr, the compiler will elide the copy when you return the std::string by value.

(Compilers are not required to by the spec but they all do it.)